servers/src/mining/stratumserver.rs

```diff
@@ -602,64 +602,72 @@ impl Handler {
 fn accept_connections(listen_addr: SocketAddr, handler: Arc<Handler>) {
 	info!("Start tokio stratum server");
 	let task = async move {
-		let mut listener = TcpListener::bind(&listen_addr).await.unwrap_or_else(|_| {
+		let listener = TcpListener::bind(&listen_addr).await.unwrap_or_else(|_| {
 			panic!("Stratum: Failed to bind to listen address {}", listen_addr)
 		});
-		let server = listener
-			.incoming()
-			.filter_map(|s| async { s.map_err(|e| error!("accept error = {:?}", e)).ok() })
-			.for_each(move |socket| {
-				let handler = handler.clone();
-				async move {
-					// Spawn a task to process the connection
-					let (tx, mut rx) = mpsc::unbounded();
-
-					let worker_id = handler.workers.add_worker(tx);
-					info!("Worker {} connected", worker_id);
-
-					let framed = Framed::new(socket, LinesCodec::new());
-					let (mut writer, mut reader) = framed.split();
-
-					let h = handler.clone();
-					let read = async move {
-						while let Some(line) = reader
-							.try_next()
-							.await
-							.map_err(|e| error!("error reading line: {}", e))?
-						{
-							let request = serde_json::from_str(&line)
-								.map_err(|e| error!("error serializing line: {}", e))?;
-							let resp = h.handle_rpc_requests(request, worker_id);
-							h.workers.send_to(worker_id, resp);
-						}
-
-						Result::<_, ()>::Ok(())
-					};
+		let server = async_stream::stream! {
+			loop {
+				match listener.accept().await {
+					Ok((socket, _)) => yield socket,
+					Err(e) => {
+						error!("accept error = {:?}", e);
+						continue;
+					}
+				}
+			}
+		}
+		.for_each(move |socket| {
```

Comment:
I can't believe how bad git is at diffing this change. It looks like it matches perfectly from the old line 611 vs this line 619 down, but git manages to mismatch it all...